### ISSUE TITLE
Yahoo Provider doesn't work anymore.. I create a new Provider Yahoo2

### ISSUE THREAD

**Issue by Fede72** [hybridauth/hybridauth#1212]
## Bug, feature or question?

Is this a bug, the Yahoo deprecated some API:

Yahoo: Signed API request has returned an error. HTTP error 410. Raw Provider API response: {"error":{"@lang":"en-US","@uri":"https://www.yahoo.com/","description":"No longer available","detail":"Social Directory APIs are no longer supported. Please visit https://developer.yahoo.com/oauth/social-directory-eol/ for more details"}}.

### Version and provider

HybrydAuth 3.3
Provider: Yahoo

### Reproduction

Simply test Yahoo Provider.

### Solution to be productized:

New Yahoo(2) Provider:

<?php
/*!
* Hybridauth
* https://hybridauth.github.io | https://github.com/hybridauth/hybridauth
*  (c) 2017 Hybridauth authors | https://hybridauth.github.io/license.html
*/

namespace Hybridauth\Provider;

use Hybridauth\Adapter\OAuth2;
use Hybridauth\Exception\UnexpectedApiResponseException;
use Hybridauth\Data;
use Hybridauth\User;

```php
/**
 * Yahoo OAuth2 provider adapter.
 */
class Yahoo2 extends OAuth2
{
    /**
    * {@inheritdoc}
    */
    protected $scope = 'sdpp-w';

    /**
    * {@inheritdoc}
    */
    protected $apiBaseUrl = 'https://api.login.yahoo.com/openid/v1/';

    /**
    * {@inheritdoc}
    */
    protected $authorizeUrl = 'https://api.login.yahoo.com/oauth2/request_auth';

    /**
    * {@inheritdoc}
    */
    protected $accessTokenUrl = 'https://api.login.yahoo.com/oauth2/get_token';

    /**
    * {@inheritdoc}
    */
    protected $apiDocumentation = 'https://developer.yahoo.com/oauth2/guide/';

    /**
    * Currently authenticated user
    */
    protected $userId = null;

    /**
    * {@inheritdoc}
    */
    protected function initialize()
    {
        parent::initialize();

        $this->tokenExchangeHeaders = [
            'Authorization' => 'Basic ' . base64_encode($this->clientId .  ':' . $this->clientSecret)
        ];
    }

    /**
     * Returns current user id
     *
     * @return int
     * @throws \Hybridauth\Exception\HttpClientFailureException
     * @throws \Hybridauth\Exception\HttpRequestFailedException
     * @throws \Hybridauth\Exception\InvalidAccessTokenException
     * @throws \Hybridauth\Exception\UnexpectedApiResponseException
     */
    protected function getCurrentUserId()
    {
        if ($this->userId) {
            return $this->userId;
        }

        $response = $this->apiRequest('userinfo', 'GET', [ 'format' => 'json']);

        $data = new Data\Collection($response);
        
        if (! $data->get('sub')) {
            throw new UnexpectedApiResponseException('Provider API returned an unexpected response userid.');
        }

        return $this->userId =  $data->get('sub');
    }

    /**
    * {@inheritdoc}
    */
    public function getUserProfile()
    {
        // Retrieve current user guid if needed
        $this->getCurrentUserId();

        $response = $this->apiRequest('/userinfo', 'GET', [ 'format' => 'json']);

        $data = new Data\Collection($response);

        if (! $data->exists('sub')) {
            throw new UnexpectedApiResponseException('Provider API returned an unexpected response profile.');
        }
        
        $userProfile = new User\Profile();

        $userProfile->identifier    = $data->get('sub');
        $userProfile->firstName     = $data->get('given_name');
        $userProfile->lastName      = $data->get('family_name');
        $userProfile->displayName   = $data->get('nickname');
        $userProfile->photoURL      = $data->get('picture');
        $userProfile->profileURL    = $data->filter('profile_images')->get('image192');
        $userProfile->language      = $data->get('lang');
        $userProfile->address       = $data->get('location');
        $userProfile->email 	    = $data->get('email');
        $userProfile->emailVerified = $data->get('email_verified');
        $userProfile->gender 		= $data->get('gender');        

        return $userProfile;
    }
}

```

**Comment by Fede72** [comment]
Ok thx, I will wait the release, for now I will use my custom Provider..

**Comment by gonzalesc** [comment]
Hi,
How can I download the 3.5 using composer?

This code doesn't work.

`{
	"repositories": [
        {
          "type": "git",
          "url": "https://github.com/hybridauth/hybridauth.git"
        }
    ],
    "require": {
        "hybridauth/hybridauth": "dev-master"
    }
}`

Regards

**Comment by ApacheEx** [comment]
I will try to release 3.5 version during the weekend

**Comment by ApacheEx** [comment]
@Fede72 released as [3.5](https://github.com/hybridauth/hybridauth/releases/tag/3.5.0)

**Comment by Fede72** [comment]
thx. It works!
